### PR TITLE
Mttl 792 update person event

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket
[Jira Ticket ](https://hackney.atlassian.net/jira/software/projects/MTTL/boards/47?selectedIssue=MTTL-792)

## Describe this PR

Raise a PersonUpdatedEvent

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test the complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: PersonApi/V1/Factories/PersonSnsFactory.cs [hunk 1]
```diff
@@ -15,10 +15,31 @@ public PersonSns Create(Person person, Token token)
...
+            return new PersonSns
+            {
+                CorrelationId = Guid.NewGuid(),
+                DateTime = DateTime.UtcNow,
+                EntityId = person.Id,
+                Id = Guid.NewGuid(),
+                EventType = UpdatePersonConstants.EVENTTYPE,
+                Version = UpdatePersonConstants.V1VERSION,
+                SourceDomain = UpdatePersonConstants.SOURCEDOMAIN,
+                SourceSystem = UpdatePersonConstants.SOURCESYSTEM,
```
Comment (evak6979) [inline PersonApi/V1/Factories/PersonSnsFactory.cs:42]: You dont need to have different update and create constants. A single place for constants is fine.

File: PersonApi/V1/Gateways/PersonSnsGateway.cs [hunk 2]
```diff
@@ -32,5 +32,18 @@ public async Task Publish(PersonSns personSns)
...
             await _amazonSimpleNotificationService.PublishAsync(request).ConfigureAwait(false);
         }
+
+        public async Task UpdatePersonPublish(PersonSns personSns)
+        {
+            string message = JsonConvert.SerializeObject(personSns);
+            var request = new PublishRequest
+            {
+                Message = message,
+                TopicArn = Environment.GetEnvironmentVariable("UPDATE_PERSON_SNS_ARN"),
```
Comment (evak6979) [inline PersonApi/V1/Gateways/PersonSnsGateway.cs:42]: This is the same as create person sns arn

File: terraform/staging/main.tf [hunk 3]
```diff
@@ -48,4 +48,10 @@ resource "aws_ssm_parameter" "new_person_sns_arn" {
   name  = "/sns-topic/staging/person_created/arn"
   type  = "String"
   value = aws_sns_topic.person_topic.arn
-}
\ No newline at end of file
+}
+
+resource "aws_ssm_parameter" "person_updated_sns_arn" {
```
Comment (evak6979) [inline terraform/staging/main.tf:53]: All these are unecessary. The ARN is the same for all of them.

File: PersonApi.Tests/AwsIntegrationTests.cs [hunk 4]
```diff
@@ -70,6 +71,21 @@ private void CreateSnsTopic()
 
             Environment.SetEnvironmentVariable("NEW_PERSON_SNS_ARN", response.TopicArn);
         }
+
+        public void UpdateSnsTopic()
```
Comment (evak6979) [inline PersonApi.Tests/AwsIntegrationTests.cs:75]: Not necessary. You can use existing topic.

**Review by evak6979 — CHANGES_REQUESTED** [review]

File: PersonApi/V1/Factories/PersonSnsFactory.cs (hunk 1 above)
Comment (humulla) [reply PersonApi/V1/Factories/PersonSnsFactory.cs:42]: But the value for the EventType is different, that's why you need different constants

**Review by evak6979 — APPROVED** [review]

**Pull request merged**